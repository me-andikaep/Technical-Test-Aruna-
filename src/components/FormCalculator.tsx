import React, { useState } from 'react';
import { sum, substract, mul, div } from './Operator';
import { FormatValue } from './FormatValue';

interface Props {}
interface Value {
	data: string | number;
	check: boolean;
}

const FormCalculator = (props: Props) => {
	const [input1, setInput1] = useState<Value>({ data: '10', check: false });
	const [input2, setInput2] = useState<Value>({ data: '2', check: false });
	const [input3, setInput3] = useState<Value>({ data: '5', check: false });
	const [result, setResult] = useState<Number | null>(null);

	function handleSubmit(action: string) {
		const data = FormatValue(input1, input2, input3);

		if (data?.length <= 1)
			return alert('angka yang diinput / di ceklis minimal 2');

		if (action === 'tambah') {
			let resultSum = data.reduce(sum);
			setResult(resultSum);
		} else if (action === 'kurang') {
			let resultSub = data.reduce(substract);
			setResult(resultSub);
		} else if (action === 'kali') {
			let resultMul = data.reduce(mul);
			setResult(resultMul);
		} else if (action === 'bagi') {
			let resultDiv = data.reduce(div);
			setResult(resultDiv);
		} else {
			return;
		}

		return;
	}

	return (
		<div className='container'>
			<form className='form' onSubmit={(e) => e.preventDefault()}>
				<div className='form-group'>
					<input
						type='text'
						className='input'
						onChange={(e) => setInput1({ ...input1, data: e.target.value })}
						value={input1.data || ''}
					/>
					<input
						type='checkbox'
						onChange={(e) =>
							setInput1((prev) => {
								return { ...prev, check: !prev.check };
							})
						}
						checked={input1.check}
					/>
				</div>
				<div className='form-group'>
					<input
						type='text'
						className='input'
						onChange={(e) => setInput2({ ...input2, data: e.target.value })}
						value={input2.data || ''}
					/>
					<input
						type='checkbox'
						onChange={(e) =>
							setInput2((prev) => {
								return { ...prev, check: !prev.check };
							})
						}
						checked={input2.check}
					/>
				</div>
				<div className='form-group'>
					<input
						type='text'
						className='input'
						onChange={(e) => setInput3({ ...input3, data: e.target.value })}
						value={input3.data || ''}
					/>
					<input
						type='checkbox'
						onChange={(e) =>
							setInput3((prev) => {
								return { ...prev, check: !prev.check };
							})
						}
						checked={input3.check}
					/>
				</div>

				<button className='btn' onClick={() => handleSubmit('tambah')}>
					+
				</button>
				<button className='btn' onClick={() => handleSubmit('kurang')}>
					-
				</button>
				<button className='btn' onClick={() => handleSubmit('kali')}>
					x
				</button>
				<button
					className='btn'
					onClick={() => handleSubmit('bagi')}
					name='bagi'
				>
					/
				</button>
			</form>

			<div className='result'>
				<h3>Hasil</h3>
				<h3>{result}</h3>
			</div>
		</div>
	);
};

export default FormCalculator;
