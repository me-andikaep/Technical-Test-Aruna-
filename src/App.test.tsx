import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
// import App from './App';
import FormCalculator from './components/FormCalculator';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const Wrapper = shallow(<FormCalculator />);

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

describe('<App />', () => {
	test('Halaman awal calculator dengan semua state null atau uncheck', async () => {});
});

test('Chek tombol bagi', () => {
	const button = shallow(<button name='bagi'>/</button>);
	button.find('button').simulate('click');
});

describe('Tampilan form kalkulator', () => {
	it('test render formnya', () => {
		render(<FormCalculator />);
	});

	it('Render Modul FormCalculator', () => {
		expect(Wrapper).toMatchSnapshot();
	});

	describe('Mount component', () => {
		describe('Event Test', () => {
			it('Klik ceklist event', () => {});
		});
	});
});