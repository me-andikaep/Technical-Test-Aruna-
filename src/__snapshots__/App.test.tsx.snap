// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tampilan form kalkulator Render Modul FormCalculator 1`] = `ShallowWrapper {}`;

exports[`Tampilan form kalkulator Render Modul Halaman 1`] = `ShallowWrapper {}`;
